lib/galaxy_test/api/test_jobs.py

```diff
@@ -95,6 +97,59 @@ def test_index_history(self, history_id):
             jobs = self.__jobs_index(data={"history_id": other_history_id})
             assert len(jobs) == 0
 
+    @uses_test_history(require_new=True)
+    def test_index_workflow_and_invocation_filter(self, history_id):
+        workflow_simple = """
+class: GalaxyWorkflow
+name: Simple Workflow
+inputs:
+  input1: data
+outputs:
+  wf_output_1:
+    outputSource: first_cat/out_file1
+steps:
+  first_cat:
+    tool_id: cat1
+    in:
+      input1: input1
+"""
+        summary = self.workflow_populator.run_workflow(workflow_simple, history_id=history_id, test_data={"input1": "hello world"},)
+        workflow_id = summary.workflow_id
```

Comment:
This is probably not the right workflow id, this is the id that corresponds to the [StoredWorkflow](https://github.com/mvdbeek/galaxy/blob/0834c9eebb025b7ee5494aebe34e54389b452dec/lib/galaxy/model/mapping.py#L979) model, while in the query you're filtering for [Workflow](https://github.com/mvdbeek/galaxy/blob/0834c9eebb025b7ee5494aebe34e54389b452dec/lib/galaxy/model/mapping.py#L997) model id.

For this test I'd suggest getting the workflow id from the invocation endpoint (`/api/invocations/<invocation_id>`).
We should refactor https://github.com/mvdbeek/galaxy/blob/0834c9eebb025b7ee5494aebe34e54389b452dec/lib/galaxy_test/base/populators.py#L999 to return both the stored_workflow_id and the workflow_id. Current it returns in stored_workflow_id as the workflow_id ... which used to be the only kind of workflow model we exposed via the API